rcl_yaml_param_parser/src/node_params.c

```diff
@@ -19,25 +19,86 @@
 rcutils_ret_t node_params_init(
   rcl_node_params_t * node_params,
   const rcutils_allocator_t allocator)
+{
+  return node_params_init_with_capacity(node_params, allocator, INIT_NUM_PARAMS_PER_NODE);
+}
+
+rcutils_ret_t node_params_init_with_capacity(
+  rcl_node_params_t * node_params,
+  const rcutils_allocator_t allocator,
+  size_t capacity)
 {
   RCUTILS_CHECK_ARGUMENT_FOR_NULL(node_params, RCUTILS_RET_INVALID_ARGUMENT);
   RCUTILS_CHECK_ALLOCATOR_WITH_MSG(
     &allocator, "invalid allocator", return RCUTILS_RET_INVALID_ARGUMENT);
+  if (capacity == 0) {
+    RCUTILS_SET_ERROR_MSG("capacity can't be zero");
+    return RCUTILS_RET_INVALID_ARGUMENT;
+  }
 
   node_params->parameter_names = allocator.zero_allocate(
-    MAX_NUM_PARAMS_PER_NODE, sizeof(char *), allocator.state);
+    capacity, sizeof(char *), allocator.state);
   if (NULL == node_params->parameter_names) {
+    RCUTILS_SET_ERROR_MSG("allocate error");
     return RCUTILS_RET_BAD_ALLOC;
   }
 
   node_params->parameter_values = allocator.zero_allocate(
-    MAX_NUM_PARAMS_PER_NODE, sizeof(rcl_variant_t), allocator.state);
+    capacity, sizeof(rcl_variant_t), allocator.state);
   if (NULL == node_params->parameter_values) {
     allocator.deallocate(node_params->parameter_names, allocator.state);
     node_params->parameter_names = NULL;
+    RCUTILS_SET_ERROR_MSG("allocate error");
+    return RCUTILS_RET_BAD_ALLOC;
+  }
+
+  node_params->capacity_params = capacity;
```

Comment:
Should we also set `node_params->num_params` to zero here?  I know it wasn't done before, but I feel like it should have been.